### ISSUE TITLE
adapt argument number of \contentsline in beamerbasetoc

### ISSUE THREAD

**Issue by u-fischer** [josephwright/beamer#698]
The next latex will change the definition of \contentsline to ensure that it always consumes 4 arguments (as it already does if hyperref does its patches).

This clashes with beamerbasetoc which writes a `\contentsline` with two arguments, relying on the current definition which only consumes one argumet if  `\l@xxx`  is not defined. So this here errors

~~~~
\makeatletter
\def\contentsline#1#2#3#4{\gdef\@contentsline@destination{#4}%
                          \csname l@#1\endcsname{#2}{#3}}
\makeatletter
\documentclass{beamer}
% from beamerbasetoc
% \renewcommand\addcontentsline[3]{\addtocontents{#1}{\protect\contentsline{#2}{#3}}}
\begin{document}
\begin{frame}
\tableofcontents
\end{frame}
\addcontentsline{toc}{section}{abc} %breaks
\addtocontents{toc}{abc}
\section{blub}
\begin{frame}
abc
\end{frame}

\end{document}
~~~~

Side remark: The definition in beamerbasetoc has be done for https://github.com/josephwright/beamer/issues/27 but I don't quite understand why this redefinition was needed. The original issue is not available, but I wonder why the user didn't simply use `\addtocontents` or gave `\l@section`  a sensible definition.

**Comment by josephwright** [comment]
So you'd suggest just dropping the line?

**Comment by u-fischer** [comment]
I think yes.  I can't see a reason why it should be needed.

**Comment by josephwright** [comment]
Fixed by 51fb6116c924246ab79fd98d99384362a2e305e9
